dev/set_matrix.py

```diff
@@ -261,7 +261,7 @@ def get_requires_python(package: str, version: str) -> str:
         ),
         None,
     )
-    candidates = ("3.8", "3.9")
+    candidates = ("3.8", "3.9", "3.10")
```

Comment:
thought this would work, but it doesn't. Can you check the `requires_python` value of xgboost?

Reply:
I think `get_requires_python` only works for released version ?

Reply:
ah you're right